soc/espressif/esp32s3/Kconfig.soc

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_ESP32S3
+	bool
+	select SOC_FAMILY_ESPRESSIF_ESP32
+	help
+	  ESP32-S3 Series
+
+# SoC with/without embedded flash
+config SOC_ESP32S3_R2
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_R2
+
+config SOC_ESP32S3_R8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_R8
+
+config SOC_ESP32S3_R8V
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_R8V
+
+config SOC_ESP32S3_FN8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_FN8
+
+config SOC_ESP32S3_PICO_N8R2
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_PICO_N8R2
+
+config SOC_ESP32S3_PICO_N8R8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_PICO_N8R8
+
+# SiP with flash and/or psram
+config SOC_ESP32S3_MINI_N8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_MINI_N8
+
+config SOC_ESP32S3_MINI_N4R2
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_MINI_N4R2
+
+config SOC_ESP32S3_WROOM_N4
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N4
+
+config SOC_ESP32S3_WROOM_N8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N8
+
+config SOC_ESP32S3_WROOM_N16
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N16
+
+config SOC_ESP32S3_WROOM_N4R8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N4R8
+
+config SOC_ESP32S3_WROOM_N8R8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N8R8
+
+config SOC_ESP32S3_WROOM_N16R8
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N16R8
+
+config SOC_ESP32S3_WROOM_N4R2
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N4R2
+
+config SOC_ESP32S3_WROOM_N8R2
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N8R2
+
+config SOC_ESP32S3_WROOM_N16R2
+	bool
+	select SOC_SERIES_ESP32S3
+	help
+	  ESP32S3_WROOM_N16R2
+
+config SOC_ESP32S3_PROCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for PROCPU (core 0).
+
+config SOC_ESP32S3_APPCPU
+	bool
+	help
+	  This hidden configuration defines that build is targeted for APPCPU (core 1).
+
+config SOC_ENABLE_APPCPU
```

Comment:
@nordicjm, I had to create this new Kconfig in order to let build system to indicate whether appcpu should be triggered and used in an AMP/Dual-core scenario. This is a bit similar to `config SECOND_CORE_MCUX`. Any considerations?

Reply:
If it just starts the other core running then that's fine